packages/compass-query-bar/test/create-mock-ai-endpoint.ts

```diff
@@ -0,0 +1,82 @@
+import http from 'http';
+import { once } from 'events';
+import type { AddressInfo } from 'net';
+
+export async function startMockAIServer(
+  {
+    response,
+    sendError,
+  }: {
+    response: any;
+    sendError?: boolean;
```

Comment:
super NIT: isn't it more flexible to have response that includes the status and control everything that way?

```
await startMockAIServer({
  response: {status: 500, json/text: {....}}
});
```

Reply:
Good thinking, will switch it over in the next pr that uses this.